p4-16/spec/P4-16-spec.mdk

```diff
@@ -4681,7 +4681,10 @@ The `return` statement immediately terminates the execution of the
 are not allowed within parsers.  `return` statements followed by an
 expression are only allowed within functions that return values; in
 this case the type of the expression must match the return type of the
-function.
+function.  Any copy-out behavior due to direction `out` or `inout`
```

Comment:
"Enclosing" is rather a lexical concept, but you want to refer to the dynamically executing action or function here. I would rather say that before transferring control to the caller the out and inout parameter values are copied to the corresponding arguments.

Reply:
The first sentence of the paragraph also uses words that look like lexically enclosing thing, too: "The `return` statement immediately terminates the execution of the `action`, `function` or `control` containing it."

Do you want to change that sentence, too?  It seems to me that the lexically enclosing action, function, or control is exactly what `return` is meant to affect, isn't it?

Reply:
Ok, you can leave the text as it is.